src/app/execution/txHelper.js

```diff
@@ -28,20 +28,20 @@ module.exports = {
   },
 
   sortAbiFunction: function (contract) {
-    var abi = JSON.parse(contract.interface).sort(function (a, b) {
-      if (a.name > b.name) {
-        return -1
-      } else {
+    return JSON.parse(contract.interface).sort(function (a, b) {
+      if (a.constant === true && b.constant !== true) {
         return 1
-      }
-    }).sort(function (a, b) {
-      if (a.constant === true) {
+      } else if (b.constant === true && a.constant !== true) {
         return -1
-      } else {
+      }
+      // If we reach here, either a and b are both constant or both not; sort by name then
+      // special case for fallback and constructor      
+      if (a.type === 'function') {
+        return a.name.localeCompare(b.name)
```

Comment:
Hm, this shouldn't be locale specific I think. Also identifiers can be ASCII only.

Reply:
`localeCompare()` is the only native string-comparison function in Javascript; would you rather multiple separate calls to check greater, less, and equal?

Reply:
Since this is only used for display purposes it should be fine, though I'd add a comment.

If this would be critical piece of code, either:
- a manual implementation
- or setting the locale to ASCII English upfront

would be needed, because certain languages consider di/trigraphs with different sorting order.